### ISSUE TITLE
Dynamic assignment is broken

### ISSUE THREAD

**Issue by joemcl** [ProgressiveCoders/Spoke#1066]
We updated to latest upstream branch main last week. Previously, with Dynamic on, we could post a link and folks could just jump in and start sending. Now, each person has to be assigned a batch and/or max count has to be updated manually from 0 to a number. cc: @deasterdaywfp

**Comment by deasterdaywfp** [comment]
Adding to this: it's not that we just have to change the max value the defaults to zero but that even upon changing and saving the texters step it will not assign correctly and therefore dynamic from the link is not working. Even upon assigning manually the dynamic setting needs to be turned off for it to work correctly.

**Comment by harpojaeger** [comment]
@shakalee14, can you add the steps you took to reproduce this locally? @joemcl, can you tell us which exact version you're on (i.e. the Git commit ID) that displays this bug and which version (the Git commit ID) you know _didn't_ display it?

**Comment by joemcl** [comment]
@harpojaeger @schuyler1d @shakalee14 FWIW - we created a staging20190219 branch here - https://github.com/WorkingFamilies/Spoke/tree/staging20190219 - via a force-push from upstream branch main...deployed that manually to our staging site, did some quick testing, then promoted that to production site via Heroku Pipeline...

**Comment by schuyler1d** [comment]
@joemcl what branch were you on before?

**Comment by joemcl** [comment]
@schuyler1d I believe this branch - https://github.com/WorkingFamilies/Spoke/tree/staging20180902 - followed same steps - deployed that manually to staging on Heroku, promoted to production on Heroku...

**Comment by joemcl** [comment]
@lperson any help with getting Git commit IDs per @harpojaeger request above - my command line Git game is still weak :( ...

**Comment by shakalee14** [comment]
Question - when a texter (who hasnt been manually added to a dynamic assignment campaign) clicks on the dynamic assignment join link (and then sees a greyed out button). Can you let me know if the the max_contacts for this user in the db (in table assignment) is being automatically set to 0?

**Comment by joemcl** [comment]
@shakalee14 yeah can do, lemme finish setting up a test campaign...

**Comment by joemcl** [comment]
OK @shakalee14 test campaign created. Selecting Dynamic for texter - I'll be the only sender. Will click on dynamic link and paste screenshot of what I see in my sender dashboard first. Campaign ID is 1196.

**Comment by joemcl** [comment]
Ok so turned on Dynamic, copied and pasted the link. Nothing appears in my sender Dashboard...

![fc6a47bc-7b39-4bd0-acbd-c54ce66f393d](https://user-images.githubusercontent.com/3912434/53527069-be160700-3a9a-11e9-917a-b6b69ccfc3e9.jpg)

**Comment by joemcl** [comment]
aargh. we had Enforce turned on. I turned that off. Refreshed. But still nothing on my Sender dashboard.

**Comment by joemcl** [comment]
Now checking if 'max_contacts for this user in the db (in table assignment) is being automatically set to 0?' ....

**Comment by joemcl** [comment]
oops. I had not started the Campaign aargh. Just did that. Now I see grayed-out on my sender Dashboard, @shakalee14 ...

![599390dc-4665-451b-9cc8-44b6202c8a49](https://user-images.githubusercontent.com/3912434/53527460-e18d8180-3a9b-11e9-95cd-d87451f7af22.jpg)

**Comment by joemcl** [comment]
*Now* checking if 'max_contacts for this user in the db (in table assignment) is being automatically set to 0?'

**Comment by joemcl** [comment]
BTW I get spinning when saving Texters, after actually starting the Campaign :) , opening up Texters to get the Dynamic link again, then saving Texters - spinning on saving that....

![b5ff48e2-240d-4f36-8362-cf646683d5f7](https://user-images.githubusercontent.com/3912434/53527771-b6eff880-3a9c-11e9-8652-46bc16510ba5.jpg) 

 @deasterdaywfp this is the other issue you were having, right? Spinning on saving Texters?

**Comment by joemcl** [comment]
aargh. still getting spinning per above. Going to select Discard Job.

**Comment by joemcl** [comment]
OK @shakalee14 after clicking Discard Job, I now see this in Texters - i.e., max contacts is 0, available to assign is 0, and of course already texted is 0...

![1fbf9383-4b88-458a-9e00-55992d1e19cc](https://user-images.githubusercontent.com/3912434/53528192-aee48880-3a9d-11e9-9a46-8a79d709b015.jpg)

**Comment by joemcl** [comment]
OK @shakalee14 in table assignment, max contacts for my user ID (I pixelated it out) is at 0... 

![198bf41a-21a3-4a2f-80ad-6aafd1817457 1](https://user-images.githubusercontent.com/3912434/53528925-f1a76000-3a9f-11e9-9e36-e915858f4e27.jpg)

**Comment by deasterdaywfp** [comment]
@shakalee14 sorry for being dense, but have you identified the PR that will fix this issue in our version/build? That is #1067 ?

**Comment by deasterdaywfp** [comment]
@joemcl @shakalee14 @harpojaeger Just checking in on this issue. Anything left to address before merging to test on our end?

**Comment by joemcl** [comment]
@shakalee14 - this is still failing in a test done 3/31/2019 in WFP staging app running in a container on Heroku... deployed this branch on our staging - https://github.com/jlegrone/Spoke/tree/review-apps

**Comment by joemcl** [comment]
Hey @shakalee14 - so when I connect to the postgres database for our staging app directly and do this - ```update assignment
set max_contacts = 1000
where id = 49;``` - then I can send using dynamic. Screenshot below. So maybe what was changed in #828 is the problem? cc: @deasterdaywfp . The update above - directly in to the database - seems to be sticking, with Dynamic on of course. But updating the max_contacts count per user and assignment directly into the database is not a solution, obviously. 

![2d1a4255-74af-4fc1-bd88-75dea59d225b](https://user-images.githubusercontent.com/3912434/55374983-4b47e380-54bf-11e9-9886-9c3a26e025cf.jpg)

**Comment by joemcl** [comment]
Also @shakalee14 will you please re-open this Issue, thanks.

**Comment by joemcl** [comment]
So here's what I figured out. Saw this in .env variables doc at https://github.com/MoveOnOrg/Spoke/blob/main/docs/REFERENCE-environment_variables.md - 

```MAX_CONTACTS_PER_TEXTER - Maximum contacts that a texter can receive. This is particularly useful for dynamic assignment. Leave it blank (which is the default value) for no maximum.```

yet we had 0, not 'blank'/empty in Heroku .env for max contacts per texter. 0 != blank/empty. 

I was  wondering if I set MAX_CONTACTS_PER_TEXTER to, say, 1000 in the Heroku app configuration/.env settings, if that will then be default for anyone clicking on Dynamic link created in a campaign - 1000 as default, rather than 0 - and allowing sending in a campaign with Dynamic enabled. 

And it worked! Dynamic works for us again. 

Setting MAX CONTACTS PER TEXTER to blank/empty also disables Dynamic, for us. So the docs need to be updated. I did a PR for that, #1082 .
